src/pyramid_tm/__init__.py

```diff
@@ -149,13 +149,14 @@ def tm_tween(request):
             if commit_veto is not None:
                 if commit_veto(request, response):
                     raise AbortWithResponse(response)
-            else:
-                # check for a squashed exception and handle it
-                # this would happen if an exception view was invoked and
-                # rendered an error response
-                exc_info = getattr(request, 'exc_info', None)
-                if exc_info is not None:
-                    maybe_tag_retryable(request, exc_info)
+
+            # check for a squashed exception and handle it
+            # this would happen if an exception view was invoked and
+            # rendered an error response
+            exc_info = getattr(request, 'exc_info', None)
+            if exc_info is not None:
+                maybe_tag_retryable(request, exc_info)
+                if commit_veto is None:
```

Comment:
I don't have all of this swapped into my brain right now, it's been quite a while since I wrote it. I'm pondering whether we should've invoked maybe_tag_retryable BEFORE all of the commit_veto cases. For example, moving that aspect to line 148 here.

Thoughts?

Reply:
I thought about it briefly, but didn't see the gain. You still have to check for the `commit_veto` because there is an explicit test to make sure it is not raising when veto is defined.

Reply:
Could we get this merged/released? 

If not, well, I could still call `maybe_tag_retryable` in our veto code, but I wouldn't think that's intended behaviour.

Reply:
unfortunately there's some work to do to fix up CICD before I can do a release

Reply:
No problem! I can still do an internal release but did not want to do that without knowing that the change is being accepted. Thanks!